test/e2e/set_delete_test.go

```diff
@@ -0,0 +1,252 @@
+}
+
+func TestSetRouteResponseHeaders(t *testing.T) {
+	t.Parallel()
+	var headerNameXFrame string = "X-Frame-Options"
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "set-response-headers-route"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, conditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+	service := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.InternalIngressControllerServiceName(ic), service); err != nil {
+		t.Fatalf("failed to get ingresscontroller service: %v", err)
+	}
+
+	echoPod := buildEchoPod("set-response-header-via-route-echo", deployment.Namespace)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", echoPod.Namespace, echoPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoPod); err != nil {
+			t.Fatalf("failed to delete pod %s/%s: %v", echoPod.Namespace, echoPod.Name, err)
+		}
+	}()
+
+	echoService := buildEchoService(echoPod.Name, echoPod.Namespace, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoService); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+		}
+	}()
+
+	echoRoute := buildRouteWithHTTPHeaders(echoPod.Name, echoPod.Namespace, echoService.Name, headerNameXFrame, "DENY")
+	if err := kclient.Create(context.TODO(), echoRoute); err != nil {
+		t.Fatalf("failed to create route %s/%s: %v", echoRoute.Namespace, echoRoute.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoRoute); err != nil {
+			t.Fatalf("failed to delete route %s/%s: %v", echoRoute.Namespace, echoRoute.Name, err)
+		}
+	}()
+
+	clientPodImage := deployment.Spec.Template.Spec.Containers[0].Image
+
+	testHeaders(t, clientPodImage, echoRoute, service.Spec.ClusterIP, nil, fmt.Sprintf("%v:", headerNameXFrame), 1)
```

Comment:
```suggestion
	testHeaders(t, clientPodImage, echoRoute, service.Spec.ClusterIP, nil, fmt.Sprintf("%v:Set:DENY", headerNameXFrame), 1)
```

Reply:
we check the curl output so it is not in the format %v:Set:DENY

Reply:
Please use the format it needs to check that the value is actually there.

Reply:
@candita currently this is what you see on failure:
`
[miheer@localhost cluster-ingress-operator]$ make TEST=TestSetIngressControllerResponseHeaders test-e2e
go generate ./pkg/manifests
CGO_ENABLED=1 GO111MODULE=on GOFLAGS=-mod=vendor go test -timeout 1h -count 1 -v -tags e2e -run "TestSetIngressControllerResponseHeaders" ./test/e2e
=== RUN   TestSetIngressControllerResponseHeaders
=== PAUSE TestSetIngressControllerResponseHeaders
=== CONT  TestSetIngressControllerResponseHeaders
    set_delete_test.go:69: failed to read output from pod set-response-header-via-ic-echo1: container "curl" in pod "set-response-header-via-ic-echo1" is waiting to start: ContainerCreating
    set_delete_test.go:69: failed to read output from pod set-response-header-via-ic-echo1: container "curl" in pod "set-response-header-via-ic-echo1" is waiting to start: ContainerCreating
    set_delete_test.go:69: failed to read output from pod set-response-header-via-ic-echo1: container "curl" in pod "set-response-header-via-ic-echo1" is waiting to start: ContainerCreating
    set_delete_test.go:83: found match 1 of 1 expected: < x-frame-options: X-Frame-Options
    panic.go:522: deleted ingresscontroller set-response-headers-ingresscontroller
--- FAIL: TestSetIngressControllerResponseHeaders (91.04s)
FAIL
FAIL	github.com/openshift/cluster-ingress-operator/test/e2e	93.644s
FAIL
make: *** [Makefile:56: test-e2e] Error 1
[miheer@localhost cluster-ingress-operator]$ 
`

Reply:
done yes